docs/dev/docs.md

```diff
@@ -66,19 +49,41 @@ Use `mkdocs gh-deploy --help` for a full list of options available for the gh-de
  
 :fa-info-circle: _**Important:**_ Never manually edit files on the `gh-pages` branch because you will lose your work the next time you run the `gh-deploy` command.
 
+
 ## Contributing to Documentation
 
+:fa-info-circle: _**Important:**_ If you added a new markdown file to `docs/`, please make sure to include it in it relevant section in `mkdocs.yml` to make it accessible.
+
+
 We use github issues to keep track of hanging fruits to improve ShapeWorks documentation. For an internal list of todos, which will be turned to issues, visit [Documentation ToDo List](../todo.md).
 
-## Inserting videos in documentation
+## Inserting Videos in Documentation
 
 [Request on GitHub](https://github.com/SCIInstitute/ShapeWorks/issues/new) for your video to be uploaded to the SCI ShapeWorks server.
 
 Insert it in the markdown file using `<p><video src="https://sci.utah.edu/~shapeworks/doc-resources/mp4s/video.mp4" autoplay muted loop controls style="width:100%"></p>`.
 
-### Where are the videos stored?
+:fa-info-circle: _**Where are the videos stored?**_ On SCI servers, at `/usr/sci/www/shapeworks/doc-resources/mp4s/`.
+
+
+
+## Auto-generating ShapeWorks Commands Documentation
+
+The `DocumentationUtils` package in `Python` has APIs for auto-documenting command-line tools and *to-come-soon* python APIs. We use the `docs` folder to save the generated documentation.
+
+
+To generate documentation for the `shapeworks` commands, use the `generateShapeWorksCommandDocumentation` function of the `DocumentationUtils` library. Be sure to run `conda_install.sh` as described in [How to Build ShapeWorks from Source?](build.md) to install `DocumentationUtils`.
+
+First, make sure that the `shapeworks` command is in your path (`set PATH=/path/to/shapeworks:$PATH`), then use Python to run the command:
+
+```python
```

Comment:
This doesn't work

```
>>> import DocumentationUtils
>>> DocumentationUtils.generateShapeWorksCommandDocumentation('docs/tools/ShapeWorksCommands.md')
----------- STDOUT -----------
Usage: shapeworks <command> [args]...

Unified ShapeWorks executable that includes command line utilities for automated construction of compact statistical landmark-based shape models of ensembles of shapes

Options:
  -h, --help                        show this help message and exit
  --version                         show program's version number and exit
  -q, --quiet                       don't print status messages

Available commands (dashes optional):
---------------------

Image
  add                           add a value to each pixel in the given image and/or add another
                                image in a pixelwise manner
  antialias                     antialiases binary volumes
  binarize                      sets portion of image greater than min and less than or equal to max
                                to the specified value
  blur                          applies gaussian blur
  bounding-box                  compute largest bounding box surrounding the specified isovalue of
                                the specified set of images
  clip                          clips volume with the specified cutting planes defined by three 3D
                                points
  close-holes                   closes holes in a volume defined by values larger than specified
                                value
  compare                       compare two images
  compute-dt                    computes signed distance transform volume from an image at the
                                specified isovalue
  crop                          crop image down to the current region (e.g., from bounding-box
                                command), or the specified min/max in each direction [default: image
                                dimensions]
  curvature                     denoises an image using curvature driven flow using curvature flow
                                image filter
  divide                        divide an image by a constant
  extract-label                 extracts/isolates a specific voxel label from a given multi-label
                                volume and outputs the corresponding binary image
  gradient                      computes gradient magnitude of an image region at each pixel using
                                gradient magnitude filter
  icp                           transform current image using iterative closest point (ICP) 3D rigid
                                registration computed from source to target distance maps
  image-to-mesh                 converts the current image to a mesh
  info                          prints requested image dimensions, spacing, size, origin, direction
                                (coordinate system), center, center of mass and bounding box
                                [default: prints everything]
  multiply                      multiply an image by a constant
  negate                        negate the values in the given image
  pad                           pads an image with specified value by specified number of voxels in
                                the x-, y-, and/or z- directions; origin remains at the same
                                location (note: negative padding to shrink an image is permitted)
  read-image                    reads an image
  recenter                      recenters an image by changing its origin in the image header to the
                                physical coordinates of the center of the image
  reflect                       reflect image with respect to logical image center and the specified
                                axis
  resample                      resamples an image using new physical spacing (computes new dims)
  resize                        resizes an image (computes new physical spacing)
  rotate                        rotates image by specified value
  scale                         scales image by specified value
  set-origin                    set origin
  sigmoid                       computes sigmoid function pixel-wise using sigmoid image filter
  subtract                      subtract a value from each pixel in this image and/or subtract
                                another image in a pixelwise manner
  topo-preserving-smooth        Helper command that applies gradient and sigmoid filters to create a
                                feature image for the TPLevelSet filter; note that a curvature flow
                                filter is sometimes applied to the image before this
  tp-levelset                   segments structures in image using topology preserving geodesic
                                active contour level set filter
  translate                     translates image by specified physical (image space) distance
  warp-image                    finds the warp between the source and target landmarks and
                                transforms image by this warp
  write-image                   writes the current image (determines type by its extension)

Mesh
  coverage                      creates mesh of coverage between two meshes
  read-mesh                     reads a mesh
  write-mesh                    writes the current mesh (determines type by its extension)

Optimize
  optimize                      generate a particle system

ParticleSystem
  compactness                   Compute compactness of a loaded particle system
  generalization                compute generalization of a loaded particle system
  read-particle-system          reads a particle system
  specificity                   compute specificity of a loaded particle system
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/karthiklocal/miniconda3/envs/shapeworks/lib/python3.7/site-packages/DocumentationUtils/__init__.py", line 24, in generateShapeWorksCommandDocumentation
    CommandsUtils.addCommand(mdFile, cmd, level = 2, spacedelim = spacedelim, verbose = True)
  File "/home/karthiklocal/miniconda3/envs/shapeworks/lib/python3.7/site-packages/DocumentationUtils/CommandsUtils.py", line 179, in addCommand
    mdFile.new_header(level = level, title = cmd.split()[-1])
  File "/home/karthiklocal/miniconda3/envs/shapeworks/lib/python3.7/site-packages/mdutils/mdutils.py", line 112, in new_header
    self.__add_new_item_table_of_content(level, title)
  File "/home/karthiklocal/miniconda3/envs/shapeworks/lib/python3.7/site-packages/mdutils/mdutils.py", line 129, in __add_new_item_table_of_content
    curr = curr[-1]
IndexError: list index out of range
```

Reply:
Also, I don't see why the user has to open a python shell and run this. It could just be a standalone command/script calls `generateShapeWorksDocs` with the path to the right md file.

Reply:
Fixed and added a script to run autodoc  without a python shell.